model/websocket_client.go

```diff
@@ -104,16 +134,38 @@ func (wsc *WebSocketClient) ConnectWithDialer(dialer *websocket.Dialer) *AppErro
 }
 
 func (wsc *WebSocketClient) Close() {
+	if wsc.closed {
+		return
```

Comment:
So this would become racy too in a way. We should not call `Close()` from different goroutines.

Reply:
Yep. This would happen if we do it from multiple goroutines. But it was already racy even for a single goroutine. I didn't want to bring atomics here. But let me know if you think it's needed.

Reply:
My only concern here is that we don't want to suddenly panic someone else's code with these changes. 

I am not aware of other projects (except load-test) that make use of this client but since it's in the main server repository that's always possibility.

Reply:
Alright, fair enough. My first PR went too far, now the new one came too far behind :P

Reply:
done